liota/dccs/iotcc.py

```diff
@@ -162,21 +162,20 @@ def on_response(msg):
                 log.debug("Processed msg: {0}".format(json_msg["type"]))
                 if json_msg["type"] == "remove_resource_response" and json_msg["body"]["result"] == "succeeded":
                     log.info("Unregistration of resource {0} with IoTCC succeeded".format(entity_obj.ref_entity.name))
+                    if entity_obj.ref_entity.entity_type != "HelixGateway":
+                        self.store_device_info(entity_obj.reg_entity_id, entity_obj.ref_entity.name,
+                                               entity_obj.ref_entity.entity_type, None, True)
+                    else:
+                        self.remove_reg_entity_details(entity_obj.ref_entity.name, entity_obj.reg_entity_id)
+                        self.store_device_info(entity_obj.reg_entity_id, entity_obj.ref_entity.name, None, None, True)
                 else:
-                    log.info("Unregistration of resource {0} with IoTCC failed".format(entity_obj.ref_entity.name))
+                    log.info("Waiting for unregistration response")
+                    on_response(self.recv_msg_queue.get(True, 20))
```

Comment:
Can we change the wait time to 300 seconds everywhere? Introduce a constant and use it please. It is really difficult to track different values at different places.

Reply:
I believe the 300 seconds was added in order to handle and tackle the situation of bulk registrations in which the IoTCC takes the significant amount of time to respond. But, that`s not the case with un-registration we are introducing wait time if in case we need to remove any stale response messages which exist in the queue.

Reply:
300 seconds safeguard is to ensure the client waits for the server to respond back for 5 minutes. If the server is heavily loaded, it will not respond back in 20 seconds.

Reply:
I have changed it to 300 the only concern I have is waiting too long for the response might not be required. Hopefully, they are no stale response messages(which are not consumed) in the Queue from HelixAdapter after refactoring, which will otherwise add the significant amount of delay as they are removed from the Queue in this method.

Reply:
I don't understand why consumption of a message, not intended as a response for this request should add "significant amount of delay". Especially, how does the values being 20 or 300 make a difference in consumption of these messages?